state-chain/pallets/cf-lending-pools/src/general_lending.rs

```diff
@@ -1055,6 +1100,7 @@ struct AssetCollateralForLoan {
 /// periodically swap collected fees into each pool's desired asset.
 pub fn lending_upkeep<T: Config>(current_block: BlockNumberFor<T>) -> Weight {
 	let config = LendingConfig::<T>::get();
+	let mut weight_used = T::DbWeight::get().reads(1);
 
 	// Collecting keys to avoid undefined behaviour in `StorageMap`
 	for borrower_id in LoanAccounts::<T>::iter_keys().collect::<Vec<_>>().iter() {
```

Comment:
Iterating the keys is expensive but not considered here.

Should probably be something like: 

```
for borrower_id in LoanAccounts::<T>::iter_keys()
  .inspect(|_| { weight_used += T::DbWeight::get().reads(1) })
  .collect::<Vec<_>>()
```